tests/test_io.py

```diff
@@ -268,6 +268,37 @@ def test_list_dir_names_errors(tmpdir):
         ([], '.', '', [], True, False),
         ([], '.', 'wav', [], False, False),
         ([], '.', 'wav', [], True, False),
+        # file
+        (
+            ['file.txt'],
+            'file.txt',
```

Comment:
Could you also add tests here where `file.txt` does only exist in a sub-folder and not in the main folder,
this should then return `['sub-folder/file.txt']`.

And a test for the case that `file1.txt` exists, but you ask for `file.txt`, which should raise an error with the current implementation.

Reply:
There is also a test missing for the case where we have a bunch of files, but the search pattern (including pattern chars like `*`) does not match any of them and we expect an empty list `[]`.

Reply:
> Could you also add tests here where file.txt does only exist in a sub-folder and not in the main folder,
this should then return ['sub-folder/file.txt'].

Well, with the current implementation, if `file.txt` does not exist then it's treated as folder and raises an error :) I added a test for it.

> There is also a test missing for the case where we have a bunch of files, but the search pattern (including pattern chars like *) does not match any of them and we expect an empty list [].

I added two more tests - one where we only match files in a sub-directory, and one where we do not match any file.